### PR TITLE
Docs `Precompiles` and `System calls`

### PR DESCRIPTION
**Pull request by jorbush**
#### Precompiles
- [x] what are precompiles?
- [x] how calls to precompiles work
- [x] use keccak as an example as it's used on every deployment (it goes to the `keccak.yul` contract which then uses the `precompile` opcode)
#### System Calls
- [x] What are system contracts?
- [x] What's a system call?
- [x] Show some examples (deployer, nonce holder, L2BaseToken) and what they're used for.

### CODE REVIEW

File: docs/general.md [hunk 1]
```diff
@@ -236,8 +236,64 @@ There are three types of `far_call`:
...
+    The address used for the `keccak256` execution is derived from the last two bytes of the current contract's address (`address_low`). If this matches the predefined `keccak256` address, the system directs the execution to the appropriate function.
+    ```rust
+    let address_bytes = vm.current_context()?.contract_address.0;
+    let address_low = u16::from_le_bytes([address_bytes[19], address_bytes[18]]);
+    if address_low == KECCAK256_ROUND_FUNCTION_PRECOMPILE_ADDRESS {
+        keccak256_rounds_function::<_, false>(0, query, heaps);
+    }
+    ```
+
+2. **Execution in `keccak.yul`**: The `keccak256_rounds_function` handles the precompile's logic. It processes the input data in blocks, applies the keccak hash algorithm, and then stores the result in the contract’s memory.
```
Comment (MarcosNicolau) [inline docs/general.md:274]: I would add that the intermediate state of the hash gets stored and not the finalize digest.

**Review by MarcosNicolau — CHANGES_REQUESTED** [review]

File: docs/general.md [hunk 2]
```diff
@@ -236,8 +236,64 @@ There are three types of `far_call`:
...
+1. **Identifying the Precompile Address**:
+
+    First, a query is created to encapsulate the data that the precompile will process.
+    ```rust
+    let query = LogQuery {
+        timestamp: Timestamp(0),
+        key: abi.to_u256(),
+        ...
+    };
+    ```
```
Comment (juan518munoz) [inline docs/general.md:264]: `LogQuery` was removed from the precompiles impl.

**Review by MarcosNicolau — APPROVED** [review]
LGTM 👍

File: docs/general.md [hunk 3]
```diff
@@ -236,8 +236,58 @@ There are three types of `far_call`:
...
+1. **Identifying the Precompile Address**: The address used for the `keccak256` execution is derived from the last two bytes of the current contract's address (`address_low`). If this matches the predefined `keccak256` address, the system directs the execution to the appropriate function.
+    ```rust
+    let address_bytes = vm.current_context()?.contract_address.0;
+    let address_low = u16::from_le_bytes([address_bytes[19], address_bytes[18]]);
+    if address_low == KECCAK256_ROUND_FUNCTION_PRECOMPILE_ADDRESS {
+        keccak256_rounds_function(abi_key, heaps)?;
+    }
+    ```
+
+2. **Execution in `keccak.yul`**: The `keccak256_rounds_function` processes the input data in blocks, applies the _Keccak hash algorithm_, and stores the intermediate hash state in the contract's memory, rather than the finalized digest.
```
Comment (gianbelinche) [inline docs/general.md:261]: I think it is not sufficiently clear that when someone needs to execute a precompile it calls a certain system contract, and that is the one that executes the opcode, in this case keccak.yul is the one executing the precompile opcode

**Review by gianbelinche — CHANGES_REQUESTED** [review]

**Review by gianbelinche — APPROVED** [review]

**Pull request merged**